### PR TITLE
nn.BatchNormalization and nn.Dropout layers from Torch

### PR DESCRIPTION
**Pull request by dkurt**
### This pullrequest changes

Non-spatial versions of batch normalization and dropout layers from Torch: `nn.BatchNormalization` and `nn.Dropout`.

resolves https://github.com/opencv/opencv/issues/10128

**Merge with extra**: https://github.com/opencv/opencv_extra/pull/413

### CODE REVIEW

**Comment by vpisarev** [comment]
:+1:
